codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -209,7 +209,14 @@ public void operationComplete(ChannelFuture future) throws Exception {
                 if (failureCause == null) {
                     // Synchronously set the headersSent flag to ensure that we do not subsequently write
                     // other headers containing pseudo-header fields.
+                    //
+                    // This is best-effort as the operation is async it may still fail later on.
```

Comment:
instead of `best-effort` maybe something like `this just sets internal stream state which is used elsewhere in the codec and doesn't necessarily mean the write will complete successfully` is more clear?

Reply:
works for me